src/ngraph/frontend/onnx_import/attribute.hpp

```diff
@@ -0,0 +1,151 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#include "ngraph/except.hpp"
+#include "onnx.pb.h"
+#include "tensor.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        // forward declaration
+        class Graph;
+
+        namespace error
+        {
+            namespace attribute
+            {
+                namespace detail
+                {
+                    struct attribute : ngraph_error
+                    {
+                        attribute(std::string msg, onnx::AttributeProto_AttributeType type)
+                            : ngraph_error{std::move(msg) + ": " +
+                                           onnx::AttributeProto_AttributeType_Name(type)}
+                        {
+                        }
+                    };
+
+                } // namespace detail
+
+                struct invalid_data : detail::attribute
```

Comment:
Unless these are mirroring onnx symbols with the same name, they should be `InvalidData`, etc.